controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -210,54 +230,70 @@ def _decode_base64_to_string(self, content_with_base64):
+        logger.info(messages.ADD_LABEL_TO_NODE.format(settings.MANAGE_NODE_LABEL, node_name))
+        self._update_manage_node_label(node_name, settings.TRUE_STRING)
 
-    def _remove_managed_by_host_definer_label(self, node_name):
+    def _remove_manage_node_label(self, node_name):
         if self._is_managed_by_host_definer_label_should_be_removed(node_name):
-            logger.info(messages.REMOVE_LABEL_FROM_NODE.format(settings.MANAGED_BY_HOST_DEFINER_LABEL, node_name))
-            self._update_node_managed_by_host_definer_label(node_name, None)
+            logger.info(messages.REMOVE_LABEL_FROM_NODE.format(settings.MANAGE_NODE_LABEL, node_name))
+            self._update_manage_node_label(node_name, None)
 
     def _is_managed_by_host_definer_label_should_be_removed(self, node_name):
         return self._is_dynamic_node_labeling_allowed() and \
             not self._is_node_has_ibm_block_csi(node_name) and \
-            self._is_node_has_managed_by_host_definer_label(node_name)
+            not self._is_node_has_host_definitions(node_name)
 
     def _is_node_has_ibm_block_csi(self, node_name):
-        csi_node = self._get_csi_node(node_name)
-        return csi_node.node_id is not None
-
-    def _define_host_on_all_storages_from_secrets(self, node_name):
+        csi_node_info = self._get_csi_node_info(node_name)
+        return csi_node_info.node_id is not None
+
+    def _is_node_has_host_definitions(self, node_name):
+        host_definitions_info = self._get_all_node_host_definitions_info(node_name)
+        return host_definitions_info is not Empty
+
+    def _get_all_node_host_definitions_info(self, node_name):
+        node_host_definitions_info = []
+        k8s_host_definitions = self._get_k8s_host_definitions()
+        for k8s_host_definition in k8s_host_definitions:
+            host_definition_info = self._generate_host_definition_info(k8s_host_definition)
+            if host_definition_info.node_name == node_name:
+                node_host_definitions_info.append(host_definition_info)
+        return node_host_definitions_info
+
+    def _define_host_on_all_storages(self, node_name):
         for secret_id, storage_classes_using_this_secret in SECRET_IDS.items():
             if storage_classes_using_this_secret == 0:
                 continue
-            host_definition = self._get_host_definition_from_secret_and_node_name(node_name, secret_id)
-            self._create_definition(host_definition)
+            host_definition_info = self._get_host_definition_info_from_secret_and_node_name(node_name, secret_id)
+            self._create_definition(host_definition_info)
+
+    def _generate_secret_id(self, secret_name, secret_namespace):
+        return (secret_name, secret_namespace)
 
-    def _generate_secret_id_from_secret_and_namespace(self, secret_name, secret_namespace):
-        return secret_name + settings.SECRET_SEPARATOR + secret_namespace
+    def _munch_watch_event(self, watch_event):
```

Comment:
`_munch` is sufficient I believe, there is no special logic for the `watch_event`..